client/src/components/PlanShowPage.js

```diff
@@ -14,18 +17,83 @@ const PlanShowPage = (props) => {
     }
   }
 
+  const fetchUser = async () => {
+    const currentUser = await getCurrentUser()
+    setUser(currentUser)
+  }
+
   useEffect(() => {
-    fetchPlan()
+    fetchPlan(),
+      fetchUser()
   }, [])
 
-  const detailComponents = Object.keys(plan).map(attr => {
-    return <p>{_.capitalize(attr)}: {plan[attr]}</p>
-  })
+  const handleClick = async (event) => {
+    try {
+      const response = await fetch(`/api/v1/signups`, {
+        method: "POST",
+        headers: new Headers({
+          "Content-Type": "application/json"
+        }),
+        body: JSON.stringify({ planId: plan.id, userId: user.id })
+      })
+      if (!response.ok) {
+        if (response.status === 422) {
+          const body = await response.json()
+          throw new Error(body.error)
+        }
+      }
+      setPlan({
+        ...plan,
+        players: [...plan.players, user]
+      })
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  const isCurrentPlayer = () => {
+    let isPlayer = false
+    plan.players.forEach(player => {
+      user.id == player.id ? isPlayer = true : null
+    })
+    return isPlayer
+  }
+
+  let playerList, playerListComponent
+  const slotsLeft = plan.playerCount - plan.players.length
+  if (plan.players.length > 0) {
+    playerList = plan.players.map(player => {
+      return <li key={player.username}>{player.username}</li>
+    })
+    playerListComponent =
+      <ul>
+        {playerList}
+        {slotsLeft
+          ? <li>{slotsLeft} spot{slotsLeft ? "s" : ""} left!</li> : ""}
```

Comment:
while this works, to most developers it would be considered poorer quality because there is too much going on for a ternary. An `if` statement would read better, and as such result in cleaner, more adaptable code